front/src/pages/back-office/add-product.jsx

```diff
@@ -0,0 +1,173 @@
+    } catch (e) {
+      setError(true)
+    }
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        const { data } = await getCategories()
+        setCategories(data)
+      } catch (e) {
+        setError(true)
+      }
+    })()
+  }, [setCategories])
+
+  if (hasError) {
+    return <p className="bg-red-700">{t("errorMessage")}</p>
+  }
+
+  return (
+    <Page page={backoffice}>
+      <NoSSR>
+        <div className="flex h-full">
+          <BackOfficeNav />
+          <section className="flex flex-col w-full px-4 pt-3 overflow-y-scroll">
+            <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={handleProductCreation}>
+              <MyForm className="text-2xl">
+                <h1 className="font-semibold text-3xl">NEW</h1>
+                <div className="flex justify-between gap-x-16">
+                  <div className="flex flex-col items-center justify-center">
+                    <UploadAndDisplayImage images={images} setImages={setImages} />
+                  </div>
+                  <div className="flex flex-1 flex-col justify-start ">
+                    <FormField
+                      name="name"
+                      type="text"
+                      placeholder={t("backoffice.placeholder.name")}
+                      label={t("backoffice.label.name") + "*"}
+                      className="py-2"
+                    />
+                    <FormField
+                      name="orderOfPriority"
+                      type="number"
+                      placeholder={t("backoffice.placeholder.orderOfPriority")}
+                      label={t("backoffice.label.orderOfPriority")}
+                      className="py-2"
+                    />
+                    <FormField
+                      name="priceWithoutTaxes"
+                      type="number"
+                      placeholder={t("backoffice.placeholder.priceWithoutTaxes")}
+                      label={t("backoffice.label.priceWithoutTaxes") + "*"}
+                      className="py-2"
+                    />
+                    <FormField
+                      name="priceWithTaxes"
+                      type="number"
+                      placeholder={t("backoffice.placeholder.priceWithTaxes")}
+                      label={t("backoffice.label.priceWithTaxes") + "*"}
+                      className="py-2"
+                    />
+                    <FormField
+                      name="materialsString"
+                      type="text"
+                      placeholder={t("backoffice.placeholder.materials")}
+                      label={t("backoffice.label.materials") + "*"}
+                      className="py-2"
+                    />
+                    <FormFieldSelect
+                      name="categoryID"
+                      placeholder={t("backoffice.placeholder.category")}
+                      label={t("backoffice.label.category")}
```

Comment:
Please rework with a `.map` to make it cleaner and factorize code

Reply:
I choosed to create a new component containing the Form rather than mapping the FormFields, as the latter ended up feeling way less readable.